pkg/deploy/generator/scripts/gatewayVMSS.sh

```diff
@@ -392,7 +392,11 @@ if [ -f \$NEW_CERT_FILE ]; then
   else
     sed -i -ne '1,/END CERTIFICATE/ p' \$NEW_CERT_FILE
   fi
-  if ! diff $NEW_CERT_FILE $CURRENT_CERT_FILE >/dev/null 2>&1; then
+
+  new_cert_sn="\$(openssl x509 -in "\$NEW_CERT_FILE" -noout -serial | awk -F= '{print \$2}')"
+  current_cert_sn="\$(openssl x509 -in "\$CURRENT_CERT_FILE" -noout -serial | awk -F= '{print \$2}')"
+  if [[ -n \$new_cert_sn ]] && [[ \$new_cert_sn != "\$current_cert_sn" ]]; then
```

Comment:
I think that `! -z` (not null or empty) is a little safer than `-n` (not null), because we may end up with just `new_cert_sn=""` somehow

Reply:
Made this change